lib/util/src/perfc.c

```diff
@@ -517,16 +532,12 @@ perfc_alloc_impl(
         goto errout;
     }
 
-    dte = dt_find(path, 1 /*exact*/);
-    if (dte) {
-        seti = (struct perfc_seti *)dte->dte_data;
-
-        /* [HSE_REVISIT] Fix me: mp_test.c abuses this... */
-        ev(seti->pcs_handle != NULL);
-
-        seti->pcs_handle = setp;
-        setp->ps_seti = seti;
-        return 0;
+    err = dt_update(path, perfc_update_dte, setp);
+    if (merr_errno(err) != ENOENT) {
+        goto errout;
+    } else {
+        /* Reset because we will allocate the new data tree element. */
+        err = 0;
```

Comment:
just put it all in one block:

```
if (err) {
    if (merr_errno(err) != ENOENT)
        goto errout;

    /* reset because... */
    err = 0;
}
```